feat: Update governance process (DRAFT)

vikas-agarwal76 opened the pull request.

jpower432 commented:
@vikas-agarwal76 @jflowers I've opened issue #136  to discuss the process for future alterations to our governance documents. To ensure transparency and minimize the impact of sweeping changes, I propose we adopt a practice of documenting individual decisions and significant changes within dedicated issues or PR descriptions. This will allow for a clearer understanding of their rationale and potential ramifications. What do you think?

degenaro commented:
> "While technical decisions for a project reside with its maintainers, the Oversight Committee may be asked to help mediate cross-project disputes or resolve decisions that a project's maintainers are unable to reach a consensus on."

@jflowers This has been included verbatim. Thx!

degenaro commented:
> * **Company Representation Limit (Removed from ~line 124):** The removal of the company representation limit is a major concern for vendor neutrality, a core CNCF principle. Nearly all successful projects enforce a limit on how many leadership seats can be held by employees of a single company to prevent any one company from having undue influence. We should re-introduce this limit; a common standard is that no more than 1/3 or 50% of the seats are held by employees of the same company.

@jflowers Maximal representation (line 150) is now incorporated, thx!

butler54 commented on GOVERNANCE.md:
```diff
@@ -6,56 +6,70 @@ The following document outlines how the OSCAL Compass project governance operate
+Changing the status of a project from non-core to core, or core to non-core will require a 2/3 majority of the oversight committee.
 
-#### Stepping Down and the Emeritus Process
+### OSCAL Compass 'lab/workgroup' projects and 'significant adopters'
```
e.g. OSCAL Compass 'lab' and 'core' projects; second section 'significant adopters'

butler54 commented on GOVERNANCE.md:
```diff
@@ -65,17 +79,92 @@ The Chairs are expected to perform a variety of functions to support Oversight C
+
+The election process is as follows:
+
+1. **Call for Nominations**: One month before the election, a call for nominations will be sent out to the OSCAL Compass community.
```
I think *how* should be added to point one.

butler54 commented on GOVERNANCE.md:
```diff
@@ -65,17 +79,92 @@ The Chairs are expected to perform a variety of functions to support Oversight C
+The election process is as follows:
+
+1. **Call for Nominations**: One month before the election, a call for nominations will be sent out to the OSCAL Compass community.
+2. **Nomination Period**: Community members will have one month to nominate candidates for the Oversight Committee. Self-nominations are permitted.
```
How again

butler54 commented on GOVERNANCE.md:
```diff
@@ -65,17 +79,92 @@ The Chairs are expected to perform a variety of functions to support Oversight C
+
+Oversight committee members are elected to serve a two year term. Members can serve two consecutive terms (4 years). Bootstrap and terms that result in equal to or less than one year served are exempt.
+
+Election cycles are scheduled such that roughly half of the seats come up for re-election each year for purposes of continuity. The exact number of seats alternates between 3 and 4.
```
Further to the point above, I think you should specify that the election cycle should be annual. it's not explicitly spelled out.

butler54 commented on GOVERNANCE.md:
```diff
@@ -108,10 +198,11 @@ In some cases, a supermajority vote is required for decision making - at least t
 Some examples include:
 
 * Carrying out [Code of Conduct](https://oscal-compass.github.io/compliance-trestle/mkdocs_code_of_conduct/) decisions requiring severe censure (supermajority of the Oversight Committee)
```
Supermajority is not defined. I believe you are saying 2/3rds but it should be explicit

butler54 commented on MEMBERSHIP.md:
```diff
@@ -65,20 +67,26 @@ Reviewers are knowledgeable about the codebase and are able review code for qual
 To become a project Reviewer, you must meet the following requirements:
 
+* Member for at least **3 months**.
+* Primary reviewer or author for at least **5 substantial PRs** to the codebase.
```
I'm not sure we want to put numbers in here yet. It might make sense to decrease each of the numbers to one and add a note at the bottom of this section that as OSCAL compass matures we expect the PR review / author count to increase.

butler54 reviewed (commented):
I've left a large number of comments here. The overall goal is to remove wiggle room for debate and or objections that a person is being excluded.

butler54 commented:
@vikas-agarwal76 looking at this document along with the PR's from @jpower432 and @jflowers it looks like this is consolidating a number of request and issues into once cohesive document.

Before voting occurs I believe we should add the closes: #issueorprnumber explicitly to the PR body to ensure the reviewer understand all the documents / issues this is intended to complete / supersede

vikas-agarwal76 commented:
@butler54 As discussed, I have incorporated your suggestions. Please see.
@jpower432 Incorporated your suggestion on keeping adopters separate from significant adopters (now termed Advocate). Also updated membership doc to allow maintainers to self nominate.

vikas-agarwal76 commented:
/vote-super
